apps/codestyle/codestyle-sql.py

```diff
@@ -187,25 +187,58 @@ def insert_delete_safety_check(file: io, file_path: str) -> None:
     file.seek(0)  # Reset file pointer to the beginning
     not_delete = ["creature_template", "gameobject_template", "item_template", "quest_template"]
     check_failed = False
-    previous_line = ""
+    delete_lines = {}
+    lines = file.readlines()
 
     # Parse all the file
-    for line_number, line in enumerate(file, start = 1):
-        if line.startswith("--"):
+    line_num = 0
+    while line_num < len(lines):
+        line = lines[line_num]
+        stripped = line.strip()
+        if stripped.startswith("--") or not stripped:
+            line_num += 1
             continue
-        if "INSERT" in line and "DELETE" not in previous_line:
-            print(f"❌ No DELETE keyword found before the INSERT in {file_path} at line {line_number}\nIf this error is intended, please notify a maintainer")
-            check_failed = True
-        previous_line = line
-        match = re.match(r"DELETE FROM\s+`([^`]+)`", line, re.IGNORECASE)
+        match = re.match(r"DELETE\s+FROM\s+`?([^`\s]+)`?", stripped, re.IGNORECASE)
```

Comment:
```suggestion
        match = re.match(r"DELETE FROM\s+`?([^`\s]+)`?", stripped, re.IGNORECASE)
```

> \s matches any whitespace character (equivalent to [\r\n\t\f\v ])

examples below should not be valid  
```
`DELETE         FROM `creature` WHERE`
```
```
DELETE
FROM `creature` WHERE
```

Reply:
Good catch sogla, when I'm back at the pc I will fix it and text it, if you see any other issues please mention :p